[quant] Support quantization of embedding lookup operators

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44217 [quant] Move EmbeddingBag eager quantization to static
* #44208 [quant] Add quantized Embedding module
* **#44207 [quant] Support quantization of embedding lookup operators**

Summary:
Use existing embedding_bag operator but set offsets to [0, 1, .. len(indices)]
Test Plan:
python test/test_quantization.py TestEmbeddingOps.test_embedding_byte

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23547385](https://our.internmc.facebook.com/intern/diff/D23547385)

supriyar committed:
[quant] Support quantization of embedding lookup operators
Summary:
Use existing embedding_bag operator but set offsets to [0, 1, .. len(indices)]
Test Plan:
python test/test_quantization.py TestEmbeddingOps.test_embedding_byte

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Support quantization of embedding lookup operators"


Summary:
Use existing embedding_bag operator but set offsets to [0, 1, .. len(indices)]
Test Plan:
python test/test_quantization.py TestEmbeddingOps.test_embedding_byte

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on test/quantization/test_quantized_op.py:
```diff
@@ -2931,6 +2930,37 @@ def test_embedding_bag_4bit_rowwise_offsets(self, num_embeddings,
+        qresult = quant_op(packed_weight, indices, sparse=False)
+
+        ref = torch.embedding(weights, indices, padding_idx=-1, scale_grad_by_freq=False, sparse=False)
+        torch.testing.assert_allclose(ref, qresult, atol=0.005, rtol=1e-3)
```
are the modified `atol` and `rtol` values only because of quantization error?

supriyar replied:
Yes

raghuramank100 commented on test/quantization/test_quantized_op.py:
```diff
@@ -2931,6 +2930,37 @@ def test_embedding_bag_4bit_rowwise_offsets(self, num_embeddings,
+        packed_weight = prepack_op(qweight)
+        qresult = quant_op(packed_weight, indices, sparse=False)
+
+        ref = torch.embedding(weights, indices, padding_idx=-1, scale_grad_by_freq=False, sparse=False)
```
If you use qweight.dequantize(), this should be an exact match

supriyar committed:
Update on "[quant] Support quantization of embedding lookup operators"


Summary:
Use existing embedding_bag operator but set offsets to [0, 1, .. len(indices)]
Test Plan:
python test/test_quantization.py TestEmbeddingOps.test_embedding_byte

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23547385](https://our.internmc.facebook.com/intern/diff/D23547385)

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Support quantization of embedding lookup operators"


Summary:
Use existing embedding_bag operator but set offsets to [0, 1, .. len(indices)]
Test Plan:
python test/test_quantization.py TestEmbeddingOps.test_embedding_byte

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23547385](https://our.internmc.facebook.com/intern/diff/D23547385)

[ghstack-poisoned]
